### ISSUE TITLE
False positive with :or destructuring with default depending on previous arg

### ISSUE THREAD

**Issue by marcomorain** [borkdude/clj-kondo#782]
**version**

`clj-kondo v2020.02.28`

**platform**

Native on macOS:

`Darwin blaster.local 19.2.0 Darwin Kernel Version 19.2.0: Sat Nov  9 03:47:04 PST 2019; root:xnu-6153.61.1~20/RELEASE_X86_64 x86_64`

**editor**

Calva

**problem**

```Clojure
(fn unresolved-symbol-x
  [x & {:keys [y]
          :or {y (inc x)}}]
  (+ y x))
```

**repro**

`echo '(fn unresolved-symbol-x [x & {:keys [y] :or {y (inc x)}}] (+ y x))' | clj-kondo --lint -`

```
marc@blaster ~/dev/circleci/circle $ echo '(fn unresolved-symbol-x [x & {:keys [y] :or {y (inc x)}}] (+ y x))' | clj-kondo --lint -
<stdin>:1:53: error: unresolved symbol x
linting took 19ms, errors: 1, warnings: 0
```

**expected behavior**

This code is valid as far as I understand, so no warnings.

**Comment by borkdude** [comment]
I'm closing this as this is undefined behavior and people should not rely on this.

<img width="1434" alt="Screenshot 2020-03-02 17 16 43" src="https://user-images.githubusercontent.com/284934/75694976-a672ef00-5ca9-11ea-9c4a-d97a552c0c76.png">

**Comment by borkdude** [comment]
Dump from another example on Slack:

```
4:56 PM
oliy Hi, me again. I've found a false positive pair in unused binding field and unresolved symbol field with this:
```
(defn foo [k {v k}] v)
(foo :a {:a 1}) ;; -> 1
``` (edited) 





new messages
4:57 PM
(both instances of k are highlighted respectively)
5:00 PM
borkdude @oliy don't do this.
https://github.com/borkdude/clj-kondo/issues/782 (edited) 
5:01 PM
oliy Ah thank you, I searched but it was hard to know how to describe the code :)
5:01 PM
borkdude If you think this is supported by Clojure (and not just accidental behavior) talk to Alex Miller or anyone else from the core team. If they think it should be supported as official Clojure, I would be happy to fix it.
5:01 PM
But as of now it seems they don't support it.
5:01 PM
oliy Yeah I find it hard to read myself so I'm not a fan of it, but it is wonderfully terse
5:01 PM
I'll just rewrite
```

**Comment by borkdude** [comment]
https://github.com/borkdude/clj-kondo/issues/673#issuecomment-675510504:

puredanger wrote:

> this is undefined and you should not do that

**Comment by borkdude** [comment]
See #916: this behavior is supported.

There are only some circumstances where this is not supported, e.g. when relying on the order in maps.
But your example was not one of them.
